### ISSUE TITLE
Option to show end time on Calendar to show 7am-9am instead of just 7am

### ISSUE THREAD

**Issue by urdhvatech** [fullcalendar/fullcalendar#2887]
Hello there,

First column on Calendar shows the time with different options like 

h(:mm)a
H:mm

But I am unable to find an option which will make time appear with end time too.

Somewhere on stack overflow I found HH:mm { - HH:mm} but that doesn't work too.

With latest version of fullCalendar too, I found nothing.

**Comment by urdhvatech** [comment]
Anyone has some ideas for me please?

**Comment by arshaw** [comment]
if you set `displayEventEnd` to `true`, you will get the desired behavior
http://fullcalendar.io/docs/text/displayEventEnd/

**Comment by urdhvatech** [comment]
Thank you for your response, but that adds event end time to Event Title.
And I am looking for an option to show end time on Axis Time - The first column on Calendar showing the time slots. 

I am looking for an option in http://fullcalendar.io/docs/agenda/slotLabelFormat/ to show the end time.

**Comment by arshaw** [comment]
oh ok. but usually the next slot underneath has the end time. so wouldn't you get something like this:

```
7:00-7:30
----------
7:30-8:00
----------
8:00-8:30
```

and isn't that a little redundant?

**Comment by urdhvatech** [comment]
Yes it looks redundant, but the user of the application needs the end time to show up like that. Is there a way to achieve that?

**Comment by arshaw** [comment]
that is currently not possible unfortunately. i can keep this ticket open in case other people need this feature

**Comment by urdhvatech** [comment]
Thank you for consideration.

**Comment by waleedarshad-vf** [comment]
Anyone working on this??? I need to this feature to add into my application as my application deals with booked appointments and we need to show slot duration from start time to end time.

**Comment by avdept** [comment]
Any news on this?

**Comment by arshaw** [comment]
no further development has happened on this.

this feature would be especially useful for [Arbitrary differing slotMinutes increments](#472).

when the API for this is tackled, best to fit in with universal naming conventions for html VS text and date VS range, [mentioned here briefly](https://github.com/fullcalendar/fullcalendar/issues/3155#issuecomment-222594680). So maybe `slotLabelRangeFormat` to differentiate

**Comment by o15a3d4l11s2** [comment]
Are there any updates on this one?

**Comment by leehan0617** [comment]
I add option like this,
```
$('#calendar').fullCalendar({
    // etcOption..
   slotLabelInterval : '00:30:00',
   viewSubSlotLabel : true,
});
```

and I change code in TimeGrid.js (start 84line) 
```
axisHtml =
				'<td class="fc-axis fc-time ' + view.widgetContentClass + '" ' + view.axisStyleAttr() + '>' +
					(isLabeled ?
						'<span>' + // for matchCellWidths
							htmlEscape(slotDate.format(this.labelFormat)) +
					 		(view.opt('viewSubSlotLabel') ? '~' + 
							 htmlEscape(slotDate.add(30, 'minutes').format(this.labelFormat)) : '') +
						'</span>' :
						''
						) +
				'</td>';
```

if you are setting timerange 40 min , 
you must set 
```
slotLabelInterval : '00:40:00',
slotDuration : '00:40:00'
```
and

```
axisHtml =
				'<td class="fc-axis fc-time ' + view.widgetContentClass + '" ' + view.axisStyleAttr() + '>' +
					(isLabeled ?
						'<span>' + // for matchCellWidths
							htmlEscape(slotDate.format(this.labelFormat)) +
					 		(view.opt('viewSubSlotLabel') ? '~' + 
							 htmlEscape(slotDate.add(40, 'minutes').format(this.labelFormat)) : '') +
						'</span>' :
						''
						) +
				'</td>';
```
more detail , I send full reqeust [https://github.com/fullcalendar/fullcalendar/pull/3498](url)

**Comment by SirDre** [comment]
Demo ref: http://www.cifullcalendar.com/v3

**Comment by jileilei** [comment]
slotLabelFormat, display a range is good options.i like this

**Comment by PHCCorso** [comment]
It'd be great to see this implemented. This would actually help me _a lot_, since the users of my application are asking for it.

**Comment by mbvb1223** [comment]
Yeah, my customer is asking me to do this feature :D

**Comment by mbvb1223** [comment]
![image](https://user-images.githubusercontent.com/11681514/85819664-9920c980-b79e-11ea-97f6-28a5b978a269.png)
```
calendar.setOption('slotLabelFormat', function (data) {
      return moment(data.date).format("HH:mm") + " ~ " + moment(data.date).add(60, 'minutes').format("HH:mm");
});
```

**Comment by vedantrajkul** [comment]
Hi I need this features.Please help

**Comment by timgonzo** [comment]
> 

Thank you! This was just what I needed.

**Comment by timgonzo** [comment]
> ![image](https://user-images.githubusercontent.com/11681514/85819664-9920c980-b79e-11ea-97f6-28a5b978a269.png)
> 
> ```
> calendar.setOption('slotLabelFormat', function (data) {
>       return moment(data.date).format("HH:mm") + " ~ " + moment(data.date).add(60, 'minutes').format("HH:mm");
> });
> ```

Thank you! This was just what I needed.

**Comment by Mouad7181** [comment]
> data.date

Hi friend ,

Please im looking to understand how did you get the data passed into the function , please im stack . thanks you
